cpu/sam0_common/periph/i2c.c

```diff
@@ -45,365 +47,207 @@
 #define SERCOM_I2CM_CTRLA_MODE_I2C_MASTER SERCOM_I2CM_CTRLA_MODE(5)
 #endif
 
-/* static function definitions */
-static void _i2c_poweron(SercomI2cm *sercom);
-static void _i2c_poweroff(SercomI2cm *sercom);
-
-static int _start(SercomI2cm *dev, uint8_t address, uint8_t rw_flag);
+static int _start(SercomI2cm *dev, uint16_t addr);
 static inline int _write(SercomI2cm *dev, const uint8_t *data, int length);
-static inline int _read(SercomI2cm *dev, uint8_t *data, int length);
+static inline int _read(SercomI2cm *dev, uint8_t *data, int length, uint8_t stop);
 static inline void _stop(SercomI2cm *dev);
 static inline int _wait_for_response(SercomI2cm *dev, uint32_t max_timeout_counter);
+static void _i2c_poweron(i2c_t dev);
+static void _i2c_poweroff(i2c_t dev);
 
 /**
  * @brief Array holding one pre-initialized mutex for each I2C device
  */
-static mutex_t locks[] = {
-#if I2C_0_EN
-    [I2C_0] = MUTEX_INIT,
-#endif
-#if I2C_1_EN
-    [I2C_1] = MUTEX_INIT,
-#endif
-#if I2C_2_EN
-    [I2C_2] = MUTEX_INIT
-#endif
-#if I2C_3_EN
-    [I2C_3] = MUTEX_INIT
-#endif
-};
+static mutex_t locks[I2C_NUMOF];
 
-int i2c_init_master(i2c_t dev, i2c_speed_t speed)
+/**
+ * @brief   Shortcut for accessing the used I2C SERCOM device
+ */
+static inline SercomI2cm *bus(i2c_t dev)
 {
-    SercomI2cm *I2CSercom = 0;
-    gpio_t pin_scl = 0;
-    gpio_t pin_sda = 0;
-    gpio_mux_t mux;
-    uint32_t clock_source_speed = 0;
-    uint8_t sercom_gclk_id = 0;
-    uint8_t sercom_gclk_id_slow = 0;
+    return i2c_config[dev].dev;
+}
 
+void i2c_init(i2c_t dev)
```

Comment:
Why no feedback if the init was successful, I imagine it would alter the program if something failed or is there an GET_INIT_STATE somewhere?